timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimeLockServiceMetadataTest.java

```diff
@@ -146,50 +136,33 @@ public void cannotPassDownMetadataForUnknownLockDescriptors() {
     }
 
     private List<Optional<LockRequestMetadata>> getAllLockEventsMetadata() {
-        return getAllLockWatchEvents().stream()
-                .map(event -> event.accept(LOCK_WATCH_EVENT_VISITOR))
-                .flatMap(Optional::stream)
-                .map(LockEvent::metadata)
-                .collect(Collectors.toList());
+        return LockWatchIntegrationTestUtilities.extractMetadata(getAllLockWatchEvents());
     }
 
     private List<LockWatchEvent> getAllLockWatchEvents() {
         ListenableFuture<ConjureStartTransactionsResponse> responseFuture =
                 timeLockService.startTransactionsWithWatches(startTransactionsRequestWithInitialVersion);
-        return AtlasFutures.getUnchecked(responseFuture).getLockWatchUpdate().accept(LOCK_WATCH_STATE_UPDATE_VISITOR);
+        return AtlasFutures.getUnchecked(responseFuture)
+                .getLockWatchUpdate()
+                .accept(AssertSuccessVisitor.INSTANCE)
+                .events();
     }
 
     private static IdentifiedLockRequest standardRequestWithMetadata(Map<LockDescriptor, ChangeMetadata> metadata) {
         return IdentifiedLockRequest.of(metadata.keySet(), 1000, "testClient", LockRequestMetadata.of(metadata));
     }
 
-    private static final class LockWatchEventVisitor implements LockWatchEvent.Visitor<Optional<LockEvent>> {
-        @Override
-        public Optional<LockEvent> visit(LockEvent lockEvent) {
-            return Optional.of(lockEvent);
-        }
-
-        @Override
-        public Optional<LockEvent> visit(UnlockEvent unlockEvent) {
-            return Optional.empty();
-        }
-
-        @Override
-        public Optional<LockEvent> visit(LockWatchCreatedEvent lockWatchCreatedEvent) {
-            return Optional.empty();
-        }
-    }
+    private enum AssertSuccessVisitor implements LockWatchStateUpdate.Visitor<LockWatchStateUpdate.Success> {
```

Comment:
This is pretty much duplicated from `UpdateVisitors`, but that class is in another test module. We could move it to a non-test module (ugly because test code now lives among production code) or move this test class over (but then we couldn't access `LockWatchIntegrationTestUtilities`). I am not too offended by this just being here.